tmk_core/protocol/vusb/vusb.c

```diff
@@ -291,6 +292,13 @@ static void send_consumer(uint16_t data) {
 #endif
 }
 
+static void send_programmable_button(report_programmable_button_t *report) {
+#ifdef PROGRAMMABLE_BUTTON_ENABLE
+	// FIXME this is discarding buttons > 16
+    send_extra(REPORT_ID_PROGRAMMABLE_BUTTON, *report);
```

Comment:
Technically you should be able to send up to 7 data bytes without any problems (max packet size for the shared endpoint in vusb is 8, but 1 byte is consumed by the report ID). You would need to define `report_programmable_button_t` as a structure similar to `report_extra_t` which also includes the `report_id` field, and send it using `usbSetInterruptShared()`.

Reply:
The goal was to keep the changes to the low-level machinery minimal for now while doing the design discussions and at the same time having something that can be used to test the functionality.